lib/rubocop/cop/rails/find_by.rb

```diff
@@ -45,6 +46,15 @@ def autocorrect(node)
             corrector.replace(first_loc, '')
           end
         end
+
+        private
+
+        def where_with_method_as_param?(node)
+          where = node.children.find { |el| el.method?(:where) }
+          return unless where
+
+          where.arguments.size == 1 && where.arguments.first.send_type?
```

Comment:
```ruby
complex_query = some_query
User.where(complex_query).first
User.where(@complex_query).first
User.where(complex_query).first
# or some variable types
```

Could you support the above case? `complex_query` is a variable.


I think we should check argument type is `hash`. Because the cop should not add any offenses if argument is not a hash.